corehq/apps/app_manager/util.py

```diff
@@ -409,7 +409,7 @@ def get_casedb_schema(form):
     This lists all case types and their properties for the given app.
     """
     app = form.get_app()
-    base_case_type = form.get_module().case_type
+    base_case_type = form.get_module().case_type if form.requires_case() else None
```

Comment:
Yeah, I'm a bit mixed. It's kind of a dumb reason that things weren't slow before—it was excluding non-case-sharing apps and it turns out that most dummy apps happen to be non-case-sharing because that's the default when you create an app. Seems like there should be a better way than relying on that accident. That said, if we think this is a widespread problem—that is, that it's common to have many, many (dummy) apps in your domain—then it's possible the right thing to do is in fact to revert